contracts/Gauge.sol

```diff
@@ -1,79 +1,147 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.11;
+pragma solidity 0.8.13;
 
-import "./interfaces/IVirtualBalanceRewardPool.sol";
+import "./interfaces/IExtraReward.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import "@openzeppelin/contracts/utils/math/Math.sol";
 import "./interfaces/IGauge.sol";
-import "./interfaces/IVeYfiRewardPool.sol";
+import "./BaseGauge.sol";
 
 import "./interfaces/IVotingEscrow.sol";
 
-contract Gauge is IGauge {
+/** @title  Gauge stake vault token get YFI rewards
+    @notice Deposit your vault token (one gauge per vault).
+    YFI are paid based on the amount of vault tokens, the veYFI balance and the duration of the lock.
+    @dev this contract is used behind multiple delegate proxies.
+ */
+
+contract Gauge is BaseGauge, IGauge {
     using SafeERC20 for IERC20;
 
-    IERC20 public rewardToken;
     IERC20 public stakingToken;
+    //// @notice veYFI
     address public veToken;
+    //// @notice the veYFI YFI reward pool, penalty are sent to this contract.
     address public veYfiRewardPool;
-    uint256 public constant DURATION = 7 days;
-    uint256 constant MAX_LOCK = 4 * 365 * 86400;
-    uint256 constant PRECISON_FACTOR = 10**6;
```

Comment:
this is a typo, usually wouldn't bother bringing this up, but it's a crucial variable and likely to be searched or grepped.